wgpu-native/src/command/mod.rs

```diff
@@ -183,10 +188,35 @@ pub fn command_encoder_begin_render_pass(
     let mut extent = None;
     let mut barriers = Vec::new();
 
+    let limits = HUB.adapters.read()[device.adapter_id].physical_device.limits();
+    let samples_count_limit = limits.framebuffer_color_samples_count;
+
     let color_attachments =
         unsafe { slice::from_raw_parts(desc.color_attachments, desc.color_attachments_length) };
     let depth_stencil_attachment = unsafe { desc.depth_stencil_attachment.as_ref() };
 
+    let attachment_sample_counts: ArrayVec<[_; crate::device::MAX_COLOR_TARGETS]> = color_attachments
+        .iter()
+        .map(|at| view_guard[at.attachment].samples)
+        .collect();
+    let sample_count = attachment_sample_counts.get(0).cloned().unwrap_or(1);
+    assert!(sample_count & samples_count_limit != 0, "Attachment sample_count must be supported by physical device limits");
+    if attachment_sample_counts.len() != 0 {
+        for sample_count_check in attachment_sample_counts.iter() {
+            assert_eq!(*sample_count_check, sample_count, "All attachments must have the same sample_count");
+        }
+    }
+
+    let resolve_sample_counts: ArrayVec<[_; crate::device::MAX_COLOR_TARGETS]> = color_attachments
```

Comment:
Why do we need to collect those? Could just go through them and compare to 1 right away

Reply:
I will have to get the length seperately, but thats ok with me.